## Chapter 12: LINEAR MODELING USING THE F DISTRIBUTION
**Name** Jaime Cano García

## Packages Used 

In [1]:
import numpy as np
from pandas import DataFrame
from patsy import dmatrices
import statsmodels.api as sm

## Quiz

In [2]:
# Seed
np.random.seed(12)

# Sample size
n = 100

# Create data
wcc = np.round(np.random.normal(15, 5, n), 0)
crp = (wcc * 2) + np.round(np.random.normal(0, 10, n), 0)
lipase = wcc + crp + np.round(np.random.normal(2, 10, n), 0)

# DataFrame
df = DataFrame({
    'WCC': wcc,
    'CRP': crp,
    'Lipase': lipase
})

# Regression setup
y, X = dmatrices('Lipase ~ WCC + CRP', data=df)

# Fit model
model = sm.OLS(y, X).fit()

# Show results
model.summary2()

<class 'statsmodels.iolib.summary2.Summary'>
"""
                 Results: Ordinary least squares
=================================================================
Model:              OLS              Adj. R-squared:     0.759   
Dependent Variable: Lipase           AIC:                754.2907
Date:               2026-04-13 20:37 BIC:                762.1062
No. Observations:   100              Log-Likelihood:     -374.15 
Df Model:           2                F-statistic:        157.3   
Df Residuals:       97               Prob (F-statistic): 3.61e-31
R-squared:          0.764            Scale:              107.28  
-------------------------------------------------------------------
            Coef.    Std.Err.      t      P>|t|     [0.025   0.975]
-------------------------------------------------------------------
Intercept   2.3741     3.0129    0.7880   0.4326   -3.6056   8.3538
WCC         0.5426     0.2684    2.0213   0.0460    0.0098   1.0754
CRP         1.1827     0.1014   11.6636   0.0000    0.9814   1.3839
-----------------------------------------------------------------
Omnibus:              0.113        Durbin-Watson:           1.936
Prob(Omnibus):        0.945        Jarque-Bera (JB):        0.287
Skew:                 0.029        Prob(JB):                0.866
Kurtosis:             2.744        Condition No.:           96   
=================================================================
Notes:
[1] Standard Errors assume that the covariance matrix of the
errors is correctly specified.
"""

**F-statistic:**
The F-statistic is big and its p-value is below 0.05, meaning that the  means  model is statistically significant. Thus, we reject the null hypothesis and conclude that at least one predictor (WCC or CRP) is associated with Lipase.

**Coefficients and p-values:**
The intercept is not statistically significant. This is because WCC is statistically significant at the 5% level, while CRP is highly statistically significant (p < 0.001). In this case, both variables contribute to predicting Lipase, with CRP having a stronger effect.

**R² value:**
The R² value is high (around 0.76), meaning that approximately 76% of the variation in Lipase is explained by WCC and CRP. This implies that the model fits the data well.